iox_catalog/src/postgres.rs

```diff
@@ -1499,6 +1499,20 @@ LIMIT $1;"#,
             .await
             .map_err(|e| Error::SqlxError { source: e })
     }
+
+    async fn list_old_style(&mut self) -> Result<Vec<Partition>> {
```

Comment:
Let save ourselves from ourselves:

```suggestion
    async fn list_old_style(&mut self) -> Result<Vec<Partition>> {
        // Correctness: the main caller of this function, the partition bloom
        // filter, relies on all partitions being made available to it.
        //
        // This function MUST return the full set of old partitions to the
        // caller - do NOT apply a LIMIT to this query. 
        // 
        // The load this query saves vastly outsizes the load this query causes.
```